okhttp/src/main/java/okhttp3/HttpSink.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3;
+
+import java.io.Closeable;
+import java.io.IOException;
+import okio.BufferedSink;
+
+public interface HttpSink extends Closeable {
+  BufferedSink sink();
+  void headers(Headers headers) throws IOException;
```

Comment:
This looks like you only intend to support HTTP/2, but in theory everything except trailers can work fine with HTTP/1.1.

If it's HTTP/2 should it be documented as such and discuss why?

If both, should the docs specify when trailers are allowed?  e.g. check the protocol?

Reply:
As far as I know there’s no convention in HTTP/1 for servers to send the response headers before the request body has been completely received.

Reply:
Your call. But it does work the same. Browsers, proxies and client libraries all need to line up perfectly. https://serverfault.com/questions/888939/http-pipelining-producing-response-before-request-body-has-finished

Reply:
We added a comment to say we support only HTTP2 for now